mypy/checkexpr.py

```diff
@@ -3406,8 +3406,14 @@ def visit_yield_from_expr(self, e: YieldFromExpr, allow_none_return: bool = Fals
 
         # Check that the expr is an instance of Iterable and get the type of the iterator produced
         # by __iter__.
-        if isinstance(subexpr_type, AnyType):
-            iter_type = AnyType(TypeOfAny.from_another_any, source_any=subexpr_type)  # type: Type
+        if isinstance(subexpr_type, TupleType):
```

Comment:
Adding this special case looks wrong. I think instead `type_is_iterable()` should be fixed (plus any potential fallout).